### PR TITLE
Javascript unit test

### PR DESCRIPTION
**Pull request by NicolasSimmonds**
How to test:
- Generate an app, ts or js
-Run yarn test at the root and check that everything works as expected

### CODE REVIEW

File: modules/create-repay-ui/_templates/templates/javascript/Layout.ejs.t [hunk 1]
```diff
@@ -0,0 +1,32 @@
+---
```
Comment (mikesoltow) [inline modules/create-repay-ui/_templates/templates/javascript/Layout.ejs.t:1]: We don't need this file for this PR, right?

File: modules/create-repay-ui/_templates/templates/shared/app.ejs.t [hunk 2]
```diff
@@ -2,20 +2,26 @@
...
+  <StyleProvider>
+    <Suspense fallback={<Loading />}>
+      <Router>
+        <Layout default>
+          <LazyUsers path="users" />
+          <LazyHome path="/" />
+        </Layout>
+      </Router>
+    </Suspense>
+  </StyleProvider>
```
Comment (mikesoltow) [inline modules/create-repay-ui/_templates/templates/shared/app.ejs.t:24]: These changes also probably don't belong in this PR

File: modules/create-repay-ui/_templates/templates/javascript/packagejson.ejs.t [hunk 3]
```diff
@@ -22,9 +36,16 @@ to: <%=directory%>/<%=name%>/package.json
     "@repay/cactus-web": "^0.9.1"
   },
   "devDependencies": {
+    "@babel/preset-react": "^7.10.4",
```
Comment (mikesoltow) [inline modules/create-repay-ui/_templates/templates/javascript/packagejson.ejs.t:39]: As discussed in Slack, I think we should use the REPAY babel preset instead.  That would require installing both `@repay/babel-preset` and `@babel/core`.

File: modules/create-repay-ui/_templates/templates/javascript/packagejson.ejs.t [hunk 4]
```diff
@@ -10,7 +10,21 @@ to: <%=directory%>/<%=name%>/package.json
...
+      "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/__mocks__/fileMock.js",
+      "\\.(css|less)$": "identity-obj-proxy"
+    },
+    "transform": {
+      "\\.ftl$": "jest-raw-loader",
+      "^.+\\.[jt]sx?$": "babel-jest"
+    },
+    "setupFilesAfterEnv": [
+      "<rootDir>/tests/__setup__/setup.js"
+    ]
```
Comment (mikesoltow) [inline modules/create-repay-ui/_templates/templates/javascript/packagejson.ejs.t:27]: Also, as discussed in Slack, my preference would be to include all of this config in the `jest.config.js` file since we are including one.

File: modules/create-repay-ui/_templates/templates/typescript/Layout.ejs.t [hunk 5]
```diff
@@ -0,0 +1,36 @@
+---
```
Comment (mikesoltow) [inline modules/create-repay-ui/_templates/templates/typescript/Layout.ejs.t:1]: Same question about including the Layout in this PR

File: modules/create-repay-ui/_templates/templates/typescript/packagejson.ejs.t [hunk 6]
```diff
@@ -11,7 +11,20 @@ to: <%=directory%>/<%=name%>/package.json
...
+      "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/__mocks__/fileMock.js",
+      "\\.(css|less)$": "identity-obj-proxy"
+    },
+    "transform": {
+      "\\.ftl$": "jest-raw-loader",
+      "^.+\\.[jt]sx?$": "babel-jest"
+    },
+    "setupFilesAfterEnv": [
+      "<rootDir>/tests/__setup__/setup.ts"
+    ]
```
Comment (mikesoltow) [inline modules/create-repay-ui/_templates/templates/typescript/packagejson.ejs.t:27]: Same comment about moving the config to `jest.config.js`

File: modules/create-repay-ui/_templates/templates/typescript/packagejson.ejs.t [hunk 7]
```diff
@@ -24,16 +37,20 @@ to: <%=directory%>/<%=name%>/package.json
   },
   "devDependencies": {
     "@repay/eslint-config": "^3.0.0",
-    "@repay/scripts": "2.0.0",
+    "@repay/scripts": "^2.0.0",
```
Comment (mikesoltow) [inline modules/create-repay-ui/_templates/templates/typescript/packagejson.ejs.t:40]: Again, `@repay/babel-preset` and `@babel/core` will need to be installed as well.

File: modules/create-repay-ui/_templates/templates/shared/example-test.ejs.t [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+describe('Test case 1', () => {
+  test('Can render the Home component', () => {
+    const { container } = render(
+      <StyleProvider>
+        <Home />
+      </StyleProvider>
+    )
+
+    expect(container).toMatchSnapshot()
```
Comment (mikesoltow) [inline modules/create-repay-ui/_templates/templates/shared/example-test.ejs.t:18]: In an actual app, snapshot testing is less useful, and behavior testing becomes more important.  Therefore, I think an assertion such as: `expect(getByText('This is the Home Page')).toBeVisible()` would be a bit more useful in terms of showing them how to use `@testing-library/react`.  To get access to `getByText`, just replace `container` with `getByText`.

Comment (NicolasSimmonds) [reply modules/create-repay-ui/_templates/templates/shared/example-test.ejs.t]: @mikesoltow Updated this PR

File: modules/create-repay-ui/_templates/templates/typescript/packagejson.ejs.t [hunk 9]
```diff
@@ -11,7 +11,7 @@ to: <%=directory%>/<%=name%>/package.json
     "build": "repay-scripts build src/index.tsx",
     "lint": "eslint \"**/*.{js,jsx,ts,tsx}\"",
     "fmt": "yarn lint --fix",
-    "test:types": "tsc -p ./tsconfig.json --noEmit"
+    "test": "jest"
```
Comment (mikesoltow) [inline modules/create-repay-ui/_templates/templates/typescript/packagejson.ejs.t:14]: We shouldn't remove test:types, as that can be useful as well.
